### PR TITLE
Disallow deleting participation if external_invoice exists (#1165)

### PR DESCRIPTION
**Pull request by Largo**

### CODE REVIEW

**Review by rnestler — CHANGES_REQUESTED** [review]
This change seems to break some other specs:

```
rspec './spec/models/group/sektions_mitglieder/ehrenmitglied_spec.rb[1:2:1:2]' # Group::SektionsMitglieder::Ehrenmitglied behaves like Mitglied dependant destroy with Mitglied role gets hard deleted if it is not old enough
rspec './spec/models/group/sektions_touren_und_kurse/tourenleiter_spec.rb[1:1:1:2]' # Group::SektionsTourenUndKurse::Tourenleiter behaves like Mitglied dependant destroy with Mitglied role gets hard deleted if it is not old enough
rspec './spec/models/group/sektions_mitglieder/beguenstigt_spec.rb[1:2:1:2]' # Group::SektionsMitglieder::Beguenstigt behaves like Mitglied dependant destroy with Mitglied role gets hard deleted if it is not old enough
```

File: spec/models/external_invoice_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    let!(:external_invoice) { Fabricate(:external_invoice, link: participation) }
+
+    it "is prevented if associated external_invoices exist" do
+      expect { participation.destroy }.not_to change { Event::Participation.count }
+      expect(participation.errors.full_messages[0]).to eq "Datensatz kann nicht gelöscht werden, " \
+        "da abhängige Rechnungen existieren."
+    end
+
+    it "succeeds if no associated external_invoices exists" do
+      external_invoice.destroy!
```
Comment (rnestler) [inline spec/models/external_invoice_spec.rb]: No point in creating it and then destroying it.

**Review by rnestler — APPROVED** [review]

File: spec/models/external_invoice_spec.rb (hunk 1 above)
Comment (rnestler) [reply spec/models/external_invoice_spec.rb]: Fixed it.

**Comment by Largo** [comment]
Specs failen wegen der Zeitumstellung...

**Pull request merged**